test/ccmath_abs_test.cpp

```diff
@@ -28,6 +29,18 @@ void test()
     }
 }
 
+template <typename T>
+void gpp_test()
+{
+    constexpr T tol = std::numeric_limits<T>::epsilon();
+
+    static_assert(std::sin(T(0)) == 0);
+    
+    constexpr T sin_1 = boost::math::ccmath::abs(std::sin(T(-1)));
+    static_assert(sin_1 > 0);
+    static_assert(boost::math::ccmath::abs(sin_1 - 0.8414709848078965066525l) < tol);
```

Comment:
I believe this should be bitwise equal to the correctly rounded result, so `abs(sin_1 - expected) < tol` is too loose. Maybe use a hexadecimal floating point literal and check for bitwise equality?

Reply:
Nice test, btw.